src/test/java/org/wso2/carbon/connector/pcml/test/unit/AS400UtilsUnitTest.java

```diff
@@ -0,0 +1,127 @@
+                "<element1>1</element1>");
+    }
+
+    @Test(description = "Test AS400Utils preparePayload for ExceptionMessage")
+    public void testAS400UtilsPreparePayloadForException() {
+        as400Utils.preparePayload(messageContext, "This is an error Message");
+        Iterator error = messageContext.getEnvelope().getBody().getChildrenWithLocalName("errorMessage");
+        String s1 = ((OMElement) error.next()).getText();
+        Assert.assertEquals(s1, "This is an error Message");
+    }
+
+    @Test(description = "Test AS400Utils preparePayload for ExceptionMessage")
+    public void testAS400UtilsGetInputParameters() throws AS400PCMLConnectorException {
+        SynapseLog log = PowerMockito.mock(SynapseLog.class);
+        when(log.isTraceOrDebugEnabled()).thenReturn(true);
+        templateContext.getMappedValues().put(AS400Constants.AS400_PCML_PROGRAM_INPUTS,
+                "<pcml:pcmlInputs xmlns:pcml=\"pcml\">"
+                        + "<pcml:pcmlInput qualifiedName=\"Addition.inputValue1\" indices=\"1,3\">5</pcml:pcmlInput>"
+                        + "<pcml:pcmlInput qualifiedName=\"Addition.inputValue2\"><num>6</num></pcml:pcmlInput>"
+                        + "<pcml:pcmlInput>12</pcml:pcmlInput>" + "<pcml:Wrong>12</pcml:Wrong>" + "</pcml:pcmlInputs>");
+        functionStack.push(templateContext);
+        messageContext.setProperty("_SYNAPSE_FUNCTION_STACK", functionStack);
+        List<PCMLInputParam> inputParameters = as400Utils.getInputParameters(messageContext, log);
+        System.out.println(inputParameters);
+        Assert.assertEquals(inputParameters.size(), 2);
+        Assert.assertEquals(inputParameters.get(0).getQualifiedName(), "Addition.inputValue1");
+        Assert.assertEquals(inputParameters.get(0).getValue(), "5");
+        Assert.assertEquals(inputParameters.get(0).getIndices().length, 2);
+        Assert.assertEquals(inputParameters.get(1).getQualifiedName(), "Addition.inputValue2");
+        Assert.assertEquals(inputParameters.get(1).getValue(), "<num>6</num>");
+        Assert.assertNull(inputParameters.get(1).getIndices());
+    }
+
+    public MessageContext createMessageContext() throws AxisFault {
+        org.apache.axis2.context.MessageContext mc = new org.apache.axis2.context.MessageContext();
+        SynapseConfiguration config = new SynapseConfiguration();
+        SynapseEnvironment env = new Axis2SynapseEnvironment(config);
+        MessageContext messageContext = new Axis2MessageContext(mc, config, env);
+        org.apache.axiom.soap.SOAPEnvelope envelope = OMAbstractFactory.getSOAP11Factory().getDefaultEnvelope();
+        OMDocument omDoc = OMAbstractFactory.getSOAP11Factory().createOMDocument();
+        omDoc.addChild(envelope);
+        envelope.getBody().addChild(createOMElement("<a>test</a>"));
+        messageContext.setEnvelope(envelope);
+        return messageContext;
+    }
+
+    public static OMElement createOMElement(String xml) {
```

Comment:
Missing Javadoc comments here. 
Also, why duplicating the same method in all classes?